tests/src/Kernel/BaseFieldTest.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_contact_forms\Kernel;
+
+use Drupal\Tests\rdf_entity\Kernel\RdfKernelTestBase;
+use Drupal\contact\Entity\ContactForm;
+use Drupal\contact\Entity\Message;
+
+/**
+ * Tests contact form with new fields.
+ */
+class BaseFieldTest extends RdfKernelTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = [
+    'options',
+    'telephone',
+    'contact',
+    'contact_storage',
+    'rdf_skos',
+    'oe_contact_forms',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+    $this->installEntitySchema('contact_message');
+  }
+
+  /**
+   * Tests successful corporate forms message.
+   */
+  public function testCorporateFormBaseFields(): void {
+    $contact_form_id = 'oe_contact_form';
+    $contact_form = ContactForm::create(['id' => $contact_form_id]);
+    $contact_form->setThirdPartySetting('oe_contact_forms', 'is_corporate_form', TRUE);
+    $contact_form->setThirdPartySetting('oe_contact_forms', 'topics', [['topic_name' => 'Topic name', 'topic_email_address' => 'topic@emailaddress.com']]);
+    $contact_form->save();
+
+    $data = [
+      'id' => 1,
+      'contact_form' => $contact_form->id(),
+      'name' => 'example',
+      'mail' => 'admin@example.com',
+      'created' => time(),
+      'ip_address' => '127.0.0.1',
+      'subject' => 'Test subject',
+      'message' => 'Test message',
+      // Add an unsupported value.
+      'oe_topic' => 1,
+    ];
+
+    $message = Message::create($data);
+    // Validate the field.
+    $violations = $message->oe_topic->validate();
+    $this->assertNotEmpty($violations);
+
+    // Set correct values.
+    $data['oe_country_residence'] = 'http://publications.europa.eu/resource/authority/country/BEL';
+    $data['oe_telephone'] = '0123456';
+    $data['oe_topic'] = 0;
+
+    $message = Message::create($data);
+    $message->save();
+
+    /** @var \Drupal\Core\Entity\Sql\SqlContentEntityStorage $entity_type_manager */
+    $entity_type_manager = \Drupal::entityTypeManager()->getStorage('contact_message');
+    $entity_type_manager->resetCache();
+    /** @var \Drupal\contact\Entity\Message $message */
+    $message = $entity_type_manager->load($message->id());
+
+    $this->assertEquals($data['oe_country_residence'], $message->get('oe_country_residence')->getValue()['0']['target_id']);
+    $this->assertEquals($data['oe_telephone'], $message->get('oe_telephone')->getValue()['0']['value']);
+    $this->assertEquals($data['oe_topic'], $message->get('oe_topic')->getValue()['0']['value']);
+  }
+
+  /**
+   * Tests that topic fields are not available in default form.
+   */
+  public function testDefaultForm(): void {
```

Comment:
This won't be needed because you will no longer alter the form in this PR i guess.

Reply:
So as discussed, exporting form display for contact forms is not possible, so the logic of the base fields needs to be handled in hook_form_alter, thus this and the form alter remain.

I will implement the fixes in 3044 so you will be able to test in that broader scope.